vdr/didweb/manager.go

```diff
@@ -333,3 +338,74 @@ func buildDocument(newDID did.DID, doc sql.DIDDocument) (did.Document, error) {
 
 	return document, nil
 }
+
+func (m Manager) NewDocument(ctx context.Context, keyFlags didsubject.DIDKeyFlags) (*didsubject.DIDDocument, error) {
+	newDID, _ := did.ParseDID(fmt.Sprintf("%s:%s:%s", m.rootDID.String(), m.tenantPath, uuid.New()))
+	var sqlVerificationMethods []didsubject.VerificationMethod
+
+	keyTypes := []didsubject.DIDKeyFlags{didsubject.AssertionKeyUsage(), didsubject.EncryptionKeyUsage()}
+	for _, keyType := range keyTypes {
+		if keyType.Is(keyFlags) {
+			verificationMethod, err := m.NewVerificationMethod(ctx, *newDID, keyType)
+			if err != nil {
+				return nil, err
+			}
+			asJson, _ := json.Marshal(verificationMethod)
+			sqlVerificationMethods = append(sqlVerificationMethods, didsubject.VerificationMethod{
+				ID:       verificationMethod.ID.String(),
+				KeyTypes: didsubject.VerificationMethodKeyType(keyType),
+				Data:     asJson,
+			})
+		}
+	}
+
+	// Create sql.DIDDocument
+	now := time.Now().Unix()
+	sqlDoc := didsubject.DIDDocument{
+		DID: didsubject.DID{
+			ID: newDID.String(),
+		},
+		CreatedAt:           now,
+		UpdatedAt:           now,
+		Version:             0,
+		VerificationMethods: sqlVerificationMethods,
+	}
+
+	return &sqlDoc, nil
+}
+
+func (m Manager) NewVerificationMethod(ctx context.Context, controller did.DID, keyUsage didsubject.DIDKeyFlags) (*did.VerificationMethod, error) {
+	var err error
+	var verificationMethodKey nutsCrypto.Key
+	verificationMethodID := did.DIDURL{
+		DID:      controller,
+		Fragment: uuid.New().String(),
+	}
+	if keyUsage.Is(didsubject.KeyAgreementUsage) {
+		return nil, errors.New("key agreement not supported for did:web")
+		// todo requires update to nutsCrypto module
+		//verificationMethodKey, err = m.keyStore.NewRSA(ctx, func(key crypt.PublicKey) (string, error) {
+		//	return verificationMethodID.String(), nil
+		//})
```

Comment:
is this part of the larger refactor or is there an issue for this?

Reply:
#1948